docs/learning/ECSqlReference/InstanceQuery.md

```diff
@@ -0,0 +1,285 @@
+
+```sql
+    SELECT [ECInstanceId] FROM [BisCore].[Element] WHERE $->[Asset_Tag] ='COMPUTER 005'
+```
+
+| ECInstanceId  |
+| ------------- |
+| 0x8000000014c |
+
+Similarly we can read any set of properties and also filter by them
+
+```sql
+    SELECT $->[RevitId], $->[LastModifier]  FROM [BisCore].[Element] WHERE $->[Asset_Tag] ='COMPUTER 005'
+```
+
+| $ -> RevitId | $ -> LastModifier |
+| ------------ | ----------------- |
+| 381840       | kiran.patkar      |
+
+ECSql will apply a property filter on selected rows such that those instances which has at least one property out of set of instance property must exists. This improve performance.
+
+```sql
+    SELECT $->[ThisPropertyDoesNotExists] from [BisCore].[Element];
+```
+
+If `ThisPropertyDoesNotExists` does not exists in `Bis.Element` derived hierarchy then no row will be returned. ECSql filter only include rows that must have at least one instance property. If any instance does not have any instance property requested then it will will be skipped.
+
+## Accessing composite properties like `NavigationProperty`, `Point2d`, `Point3d` or `Struct`s
+
+Only top level instance property can be accessed via `$-><prop>` syntax. Doing something like `$->Model.Id` will not not work as of now. It might be supported in future but as of now any access-string within a composite property is not supported, if its the only property selected then zero row will be returned.
+
+Following type of properties can directly be use in filters and return strong type value.
+
+- DateTime
+- Integer
+- Long
+- Binary
+- String
+- Double
+
+Here is example of `RevitId` use with `IN()` clause.
+
+```sql
+    SELECT $ from [BisCore].[Element] WHERE $->RevitId In ( 1000, 2000, 3000 );
+```
+
+While composite properties are returned as `JSON`.
+
+```sql
+    SELECT $->[Model] from [RevitDynamic].[Computer] where [ECInstanceId] = 0x8000000014c;
+```
+
+above will return following
+
+| $ -> Model                                     |
+| ---------------------------------------------- |
+| `{"Id":"0x80000000003","RelECClassId":"0x51"}` |
+
+While following will not return any row
+
+```sql
+    SELECT $->[Model].[Id] from [RevitDynamic].[Computer] where [ECInstanceId] = 0x8000000014c;
+```
+
+But you can still do following to get child property
+
+```sql
+    SELECT JSON_EXTRACT($->[Model], '$.Id') AS ModelId from [RevitDynamic].[Computer] where [ECInstanceId] = 0x8000000014c;
+```
+
+above will return following
+
+| ModelId         |
+| --------------- |
+| `0x80000000003` |
+
+## Optional and non-optional instance properties
+
+By default, all properties accessed via instance accessor i.e. `$->prop` must exist in the class identifying the row for that row to qualify for output.
+
+If the user uses `?` after a property accessor e.g. `$->prop?` then it will be considered optional, and the row class will not be checked to see if the `prop` exists or not.
```

Comment:
```suggestion
If the user uses `?` after a property accessor e.g. `$->prop?` then it will be considered optional, and the row class will not be checked to see if the `prop` exists.
```